src/client/mod.rs

```diff
@@ -113,10 +84,59 @@ pub trait IotHub {
     async fn shutdown(&mut self);
 }
 
-/// iothub cloud to device (C2D) and device to cloud (D2C) messages
-mod message;
-/// client implementation, either device, module or edge
-mod twin;
+/// Trait which provides functions for communication with azure iothub
+#[async_trait(?Send)]
+pub trait IotHubBuilder {
```

Comment:
Is there a reason why the IotHubBuilder is a trait?

Reply:
the only reason is to be able to build a test IotHubClient.

Reply:
ack, thought so